src/mqt/yaqs/core/data_structures/result.py

```
@@ -153,20 +173,92 @@ def aggregate_counts(result: Result) -> None:
     result.counts = dict(sorted(counts.items()))
 
 
+def _segment_trace(
```

Ruff (0.16.1)

[warning] Too many branches (13 > 12)

(PLR0912)